source.go

```diff
@@ -131,3 +149,25 @@ func (s *Source) Teardown(ctx context.Context) error {
 	}
 	return nil
 }
+
+func (s *Source) getTableColumns(conn *pgx.Conn) ([]string, error) {
+	query := fmt.Sprintf("SELECT column_name FROM information_schema.columns WHERE table_name = '%s'", s.config.Table[0])
+
+	rows, err := conn.Query(context.Background(), query)
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+
+	var columns []string
+	for rows.Next() {
```

Comment:
We need to check `rows.Err()` after the loop, see [handling errors](http://go-database-sql.org/errors.html).